internal/gen/types/typescript/typescript.go

```diff
@@ -14,9 +15,69 @@ import (
 	"github.com/supabase/cli/internal/utils"
 )
 
-var ctx = context.Background()
+func Run(ctx context.Context, useLocal bool, useLinked bool, projectId string, dbUrl string, schemas []string, fsys afero.Fs) error {
+	coalesce := func(args ...[]string) []string {
+		for _, arg := range args {
+			if len(arg) != 0 {
+				return arg
+			}
+		}
+		return []string{}
+	}
+
+	// Generating types on `projectId` and `dbUrl` should work without `supabase
+	// init` - i.e. we shouldn't try to load the config for these cases.
+
+	if projectId != "" {
+		resp, err := utils.GetSupabase().GetTypescriptTypesWithResponse(ctx, projectId, func(_ context.Context, req *http.Request) error {
+			q := req.URL.Query()
+			q.Add("included_schemas", strings.Join(coalesce(schemas, []string{"public"}), ","))
+			req.URL.RawQuery = q.Encode()
+			return nil
+		})
+		if err != nil {
+			return err
+		}
+
+		if resp.JSON200 == nil {
+			return errors.New("failed to retrieve generated types: " + string(resp.Body))
+		}
+
+		fmt.Print(resp.JSON200.Types)
```

Comment:
```suggestion
		fmt.Println(resp.JSON200.Types)
```

any reason not to print a line break? same for a few other places

Reply:
seems like json response already has line break

Reply:
I just assumed the generated types already has it